File: dsa/array/2pointer/Parity.java
public class Parity {
    public static void main(String[] args) {
        int []arr={2,3,4,7,8};
        int left =0;
        int right=arr.length-1;
        while(left<=right){
            if(arr[left]%2==1 && arr[right]%2==0){
                int temp=arr[left];
                arr[left]=arr[right];
                arr[right]=temp;
                left++;
                right--;
            }else if(arr[left]%2==0) left++;

            else right--;
        }
        for(int i:arr){
        System.out.print(i+" ");
    }
    }
    
}

File: dsa/array/2pointer/parity.py
arr=[1,2,4,3,5,6,9,8]
left=0
right=len(arr)-1
while left<right:
    if(arr[left]%2==1 and arr[right]%2==0):
        arr[left],arr[right]=arr[right],arr[left]
        left+=1
        right-=1
    elif arr[left]%2==0:
        left+=1
    else:
        right-=1
print(arr)